In [1]:
#Imports


import pandas as pd
from pandas import json_normalize
import requests 
import json
import os
from dotenv import load_dotenv #conda install -c conda-forge python-dotenv
import time
import folium
from folium import Choropleth, Circle, Marker, Icon, Map
from folium.plugins import HeatMap, MarkerCluster
import os
import requests
from bs4 import BeautifulSoup
import pandas as pd
import re
import numpy as np
from geopy.distance import geodesic as GD

import src.functions as functions

In [2]:
import os
load_dotenv()
key = os.getenv("4squarekey")

In [3]:
df = pd.read_csv('src/locations_df.csv')

In [4]:
df.rename(columns={"Unnamed: 0": "index"}, inplace=True)


In [5]:
df.drop(columns=["index"],inplace=True)

In [6]:
# f = 100000 + design * 20/87 * 0.1 + tech * 15/87 * 0.1 - starb * 10/87 * 0.1 - airp * 20/87 *0.15 - clubs * 87/87 * 0.1 
#- vegan * 1/87* 0.05 - school * 26/87 * 0.1 - bask* 1/87*0.05 - dog * 0.05*1/87 - cost * 87/87 *0.1 - wage * 87/87 * 0.1



In [7]:
rank = []
for index,row in df.iterrows():
    formula = 10 + row["nº design comp"] * 20/87 * 0.1 + row["nº tech (>$1M)"] * 15/87 * 0.1 - row["dist to starbucks (km)"] * 10/87 * 0.1 - row["dist to airport (km)"] * 20/87 *0.15 - row["average dist to clubs (km)"] * 87/87 * 0.1 - row["average dist to vegan rest. (km)"]*1/87 * 0.05 - row["average dist to schools (km)"] * 26/87 * 0.1 - row["dist. to closes basketball court (km)"]*1/87 *0.05 - row["dist. to closest canine salon (km)"]*1/87 * 0.05 - row["cost living index"] / row["avg_wage"] * 87/87 * 0.2
    rank.append(formula)
rank
        
df["rank"] = rank
df
        
        


name       city country   latitude   longitude  \
0                      Lala  Palo Alto     USA  37.451151 -122.154369   
1                   PodTech  Palo Alto     USA  37.408256 -122.154176   
2                 SonicSwap  Palo Alto     USA  37.444487 -122.163979   
3                MyMiniLife  Palo Alto     USA  37.437328 -122.159928   
4                   Mytopia  Palo Alto     USA  34.007967 -118.489681   
..                      ...        ...     ...        ...         ...   
203             BAM Studios    Chicago     USA  41.893622  -87.627423   
204                 Plugola    Chicago     USA  41.907542  -87.662990   
205            Midway Games    Chicago     USA  41.879535  -87.624333   
206  Blizzard Entertainment     Irvine     USA  33.669500 -117.822200   
207            DanceShowoff     Irvine     USA  33.671428 -117.791423   

     nº design comp  nº tech (>$1M)  dist to starbucks (km)  \
0                 6               4                   0.729   
1                 6               4                   2.312   
2                 6               4                   0.396   
3                 6               4                   1.046   
4                 6               4                   0.357   
..              ...             ...                     ...   
203               8               2                   0.231   
204               8               2                   0.540   
205               8               2                   0.025   
206               1               1                   0.743   
207               1               1                   0.193   

     dist to airport (km)  average dist to clubs (km)  \
0                  27.550                    4.787222   
1                  30.953                   10.432143   
2                  27.437                    2.177375   
3                  28.251                    2.589750   
4                  10.506                    4.055222   
..                    ...                         ...   
203                24.437                    0.731778   
204                21.131                    1.645100   
205                25.302                    0.740111   
206                 3.813                    3.196300   
207                 6.572                    3.419700   

     average dist to vegan rest. (km)  average dist to schools (km)  \
0                            9.683300                      2.134000   
1                           12.284100                      2.422900   
2                            9.110444                      1.773333   
3                           10.697556                      1.199333   
4                            2.979500                      2.051400   
..                                ...                           ...   
203                          0.989333                      0.709700   
204                          1.600250                      1.271100   
205                          0.807111                      0.488400   
206                          4.863700                      1.819300   
207                          5.065500                      5.104000   

     dist. to closes basketball court (km)  \
0                                    1.917   
1                                    3.002   
2                                    1.115   
3                                    0.304   
4                                    0.759   
..                                     ...   
203                                  0.891   
204                                  0.964   
205                                  0.565   
206                                  0.587   
207                                  0.413   

     dist. to closest canine salon (km)  cost living index  avg_wage      rank  
0                                 6.491              95.43      6349  8.692617  
1                                 6.842              95.43      6349  7.981631  
2                                 5.767              95

In [8]:
df.sort_values(by=['rank'], ascending= False, inplace=True)
df

name           city country   latitude   longitude  \
112  Unity Technologies  San Francisco     USA  55.692461   12.530107   
62             Babelgum         London     GBR  53.344104   -6.267494   
94               Sutros  San Francisco     USA  37.756922 -122.420870   
87             DanceJam  San Francisco     USA  37.781557 -122.407959   
173              Vibeex          Paris     FRA  48.856667    2.350987   
..                  ...            ...     ...        ...         ...   
145    Verismo Networks  Mountain View     USA  37.387434 -122.051793   
9                Veeple      Palo Alto     USA  37.421142 -122.101182   
142            PicksPal  Mountain View     USA  37.400470 -122.072981   
143               Raptr  Mountain View     USA  37.403840 -122.078080   
140              PeerMe  Mountain View     USA  37.419819 -122.090188   

     nº design comp  nº tech (>$1M)  dist to starbucks (km)  \
112              12              24                   2.726   
62               16               3                   0.852   
94               12              24                   1.143   
87               12              24                   0.548   
173               4               2                   0.293   
..              ...             ...                     ...   
145               1               2                   1.928   
9                 6               4                   0.428   
142               1               2                   0.615   
143               1               2                   0.129   
140               1               2                   0.997   

     dist to airport (km)  average dist to clubs (km)  \
112                10.072                    2.213200   
62                  9.321                    0.363900   
94                 15.857                    1.058300   
87                 18.404                    0.466333   
173                 0.194                    0.878778   
..                    ...                         ...   
145                39.023                    6.268000   
9                  33.275                   10.879250   
142                36.661                    7.848750   
143                36.075                    9.042875   
140                34.109                   10.682250   

     average dist to vegan rest. (km)  average dist to schools (km)  \
112                          1.152200                      1.660700   
62                           0.532100                      0.785700   
94                           7.321222                      1.174000   
87                           6.389200                      0.670000   
173                          1.172667                      2.073286   
..                                ...                           ...   
145                         11.550200                      3.695900   
9                           10.756800                      1.732500   
142                          9.541300                      3.034800   
143                          9.857300                      2.972200   
140                         10.375000                      1.963700   

     dist. to closes basketball court (km)  \
112                                  0.806   
62                                   1.713   
94                                   1.762   
87                                   0.539   
173                                  0.035   
..                                     ...   
145                                  1.389   
9                                    0.264   
142                                  0.161   
143                                  0.743   
140                                  1.169   

     dist. to closest canine salon (km)  cost living index  avg_wage  \
112                               0.215              95.43      6349   
62                                0.105              77.87      3497   
94                                3.547              95.43      6349   
87    

### There are coordenates that don't meet the city --> I will remove these ones


In [9]:
real_country=[]
for index,row in df.iterrows():
    real_country.append(functions.check_coord(row["latitude"],row["longitude"]))
  

[['DK'],
 ['IE'],
 ['US'],
 ['US'],
 ['FR'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['GB'],
 [],
 [],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 [],
 ['US'],
 ['US'],
 ['US'],
 [],
 [],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['FR'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['CA'],
 ['CA'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['US'],
 ['CA'],
 ['US'],
 ['CA'],
 ['CA'],
 ['CA'],
 ['CA'],
 ['US'],
 ['NL'],
 ['US'],
 ['NL'],
 ['GB'],
 ['NL'],
 ['GB'],
 [],
 ['US'],
 ['US'],
 ['US'],
 ['GB'],
 ['US'],
 ['US'],
 ['GB'],
 ['GB'],
 ['GB'],
 ['GB'],
 ['US'],
 [],
 ['US'],
 [],
 ['CA'],
 ['US'],
 ['US'

In [11]:
a = df["country"]
prova = pd.DataFrame({"country DB":a, "country 4Sq":real_country})

In [12]:
prova.head(10)

country DB country 4Sq
112        USA        [DK]
62         GBR        [IE]
94         USA        [US]
87         USA        [US]
173        FRA        [FR]
86         USA        [US]
79         USA        [US]
84         USA        [US]
110        USA        [US]
99         USA        [US]

#### I remove the two first companies, my locations do not match the city, nor country, so they will be excluded from the analysis


In [13]:
df = df.iloc[2:]
df

name           city country   latitude   longitude  \
94             Sutros  San Francisco     USA  37.756922 -122.420870   
87           DanceJam  San Francisco     USA  37.781557 -122.407959   
173            Vibeex          Paris     FRA  48.856667    2.350987   
86         Kongregate  San Francisco     USA  37.786942 -122.401245   
79               Kyte  San Francisco     USA  37.788482 -122.409173   
..                ...            ...     ...        ...         ...   
145  Verismo Networks  Mountain View     USA  37.387434 -122.051793   
9              Veeple      Palo Alto     USA  37.421142 -122.101182   
142          PicksPal  Mountain View     USA  37.400470 -122.072981   
143             Raptr  Mountain View     USA  37.403840 -122.078080   
140            PeerMe  Mountain View     USA  37.419819 -122.090188   

     nº design comp  nº tech (>$1M)  dist to starbucks (km)  \
94               12              24                   1.143   
87               12              24                   0.548   
173               4               2                   0.293   
86               12              24                   0.107   
79               12              24                   0.399   
..              ...             ...                     ...   
145               1               2                   1.928   
9                 6               4                   0.428   
142               1               2                   0.615   
143               1               2                   0.129   
140               1               2                   0.997   

     dist to airport (km)  average dist to clubs (km)  \
94                 15.857                    1.058300   
87                 18.404                    0.466333   
173                 0.194                    0.878778   
86                 18.952                    0.517800   
79                 19.181                    0.301667   
..                    ...                         ...   
145                39.023                    6.268000   
9                  33.275                   10.879250   
142                36.661                    7.848750   
143                36.075                    9.042875   
140                34.109                   10.682250   

     average dist to vegan rest. (km)  average dist to schools (km)  \
94                           7.321222                      1.174000   
87                           6.389200                      0.670000   
173                          1.172667                      2.073286   
86                           6.841111                      0.565300   
79                           6.435700                      0.952400   
..                                ...                           ...   
145                         11.550200                      3.695900   
9                           10.756800                      1.732500   
142                          9.541300                      3.034800   
143                          9.857300                      2.972200   
140                         10.375000                      1.963700   

     dist. to closes basketball court (km)  \
94                                   1.762   
87                                   0.539   
173                                  0.035   
86                                   0.826   
79                                   0.159   
..                                     ...   
145                                  1.389   
9                                    0.264   
142                                  0.161   
143                                  0.743   
140                                  1.169   

     dist. to closest canine salon (km)  cost living index  avg_wage      rank  
94                                3.547              95.43      6349  9.978544  
87                                2.387              95.43      6349  9.973720  
173                               0.093              70.75      2511  9.9

### My cities are:
    1. San Francisco
    2. San Francisco
    3. Paris
    4. San Francisco
    5. San Francisco

### Option 1: Sutros, San Francisco, USA, lat=37.756922, long=-122.420870

### Option 2: DanceJam, San Francisco, USA, lat = 37.781557, long = -122.407959

### Option 3: Vibeex, Paris, FRA, lat = 48.856667, long= 2.350987	